cognite/client/_api/documents.py

```diff
@@ -0,0 +1,672 @@
+
+        Only the 100 first pages will be included.
+
+        Previews will be rendered if necessary during the request. Be prepared for the request to take a few seconds to complete.
+
+        Args:
+            id: The server-generated ID for the document you want to retrieve the preview of.
+
+        Returns:
+            bytes: The pdf preview of the document.
+
+        Examples:
+
+        Download PDF preview of file with id 123:
+
+            >>> from cognite.client import CogniteClient
+            >>> c = CogniteClient()
+            >>> content = c.documents.preview.download_pdf_bytes(id=123)
+        """
+        res = self._do_request("GET", f"{self._RESOURCE_PATH}/{id}/preview/pdf", accept="application/pdf")
+        return res.content
+
+    def download_pdf_to_path(self, path: Path | str, id: int) -> None:
+        """`Downloads a pdf preview of the specified document. <https://developer.cognite.com/api#tag/Document-preview/operation/documentsPreviewPdf>`_
+
+        Only the 100 first pages will be included.
+
+        Previews will be rendered if necessary during the request. Be prepared for the request to take a few seconds to complete.
+
+        Args:
+            path: The path to save the pdf preview of the document. If the path is a directory, the
+                  file name will be '[id].pdf'.
+            id: The server-generated ID for the document you want to retrieve the preview of.
+
+        Returns:
+            bytes: The pdf preview of the document.
+
+        Examples:
+
+        Download PDF preview of file with id 123 to folder "previews":
+
+            >>> from cognite.client import CogniteClient
+            >>> c = CogniteClient()
+            >>> c.documents.preview.download_pdf_to_path("previews", id=123)
+
+        """
+        path = Path(path)
+        if path.is_dir():
+            path = path / f"{id}.pdf"
+        else:
+            if path.suffix != ".pdf":
+                raise ValueError("Path must be a directory or end with .pdf")
+        content = self.download_pdf_bytes(id)
+        path.write_bytes(content)
+
+    def retrieve_pdf_link(self, id: int) -> TemporaryLink:
+        """`Retrieve a Temporary link to download pdf preview <https://developer.cognite.com/api#tag/Document-preview/operation/documentsPreviewPdfTemporaryLink>`_
+
+        Args:
+            id: The server-generated ID for the document you want to retrieve the preview of.
+
+        Returns:
+            A temporary link to download the pdf preview.
+
+        Examples:
+
+        Retrieve the PDF preview download link for document with id 123:
+
+            >>> from cognite.client import CogniteClient
+            >>> c = CogniteClient()
+            >>> link = c.documents.preview.retrieve_pdf_link(id=123)
+        """
+        res = self._get(f"{self._RESOURCE_PATH}/{id}/preview/pdf/temporarylink")
+        return TemporaryLink.load(res.json())
```

Comment:
Not sure if I follow you here. Are you suggesting a tuple instead of a data class for the return type? in that case I prefer the data class as it gives names to the two types returned.